vpr/SRC/route/route_timing.c

```diff
@@ -514,27 +561,110 @@ bool timing_driven_route_net(int inet, int itry, float pres_fac, float max_criti
+		pathfinder_update_one_cost(trace_head[inet], -1, pres_fac);
+		free_traceback(inet);
+
+		// if entirely pruned, have just the source (not removed from pathfinder costs)
+		if (destroyed) {
+#ifdef PROFILE
+			++entire_tree_pruned;
+			// traceback remains empty for just the root and no need to update pathfinder costs
+#endif
+		}
+		else {
+#ifdef PROFILE
+			++part_tree_preserved;
+#endif
+			// sync traceback into a state that matches the route tree
+			traceback_from_route_tree(inet, rt_root, num_sinks - remaining_targets.size());
+			// put the updated costs of the route tree nodes back into pathfinder
+			pathfinder_update_one_cost(trace_head[inet], 1, pres_fac);
+		}
+
+		// give lookup on the reached sinks
+		put_sink_rt_nodes_to_pins_lookup(rr_node_to_pin[inet], reached_sinks, rt_node_of_sink);
+
+#ifdef PROFILE
+
+		unsigned int bin {num_sinks / fanout_per_bin};
+		unsigned int sinks_left_to_route = remaining_targets.size();
+		unsigned int sinks_already_routed = num_sinks - remaining_targets.size();
+		float rebuild_time {static_cast<float>(clock() - rebuild_start) / CLOCKS_PER_SEC};
+
+		finished_sinks[bin] += sinks_already_routed;
+		rerouted_sinks[bin] += sinks_left_to_route;
+		time_on_fanout_rebuild[bin] += rebuild_time;
+#endif
+		// check for R_upstream C_downstream and edge correctness
+		INCREMENTAL_REROUTING_ASSERT(is_valid_route_tree(rt_root));
+		// congestion should've been pruned away
+		INCREMENTAL_REROUTING_ASSERT(is_uncongested_route_tree(rt_root));
+
+		// use the nodes to directly mark ends before they get converted to pins
+		mark_remaining_ends(inet, remaining_targets);
+
+		// everything dealing with a net works with it in terms of its sink pins; need to convert its sink nodes to sink pins
+		convert_sink_node_to_pins_of_net(rr_node_to_pin[inet], remaining_targets);
+
+		// still need to calculate the tree's time delay (0 Tarrival means from SOURCE)
+		load_route_tree_Tdel(rt_root, 0);
+		// mark the lookup (rr_node_route_inf) for existing tree elements as NO_PREVIOUS so add_to_path stops when it reaches one of them
+		load_route_tree_rr_route_inf(rt_root);
+	}
+	// after this point the route tree is correct
+	// remaining_targets from this point on are the **pin indices** that have yet to be routed 
+
+	// should always have targets to route to since some parts of the net are still congested
+	INCREMENTAL_REROUTING_ASSERT(!remaining_targets.empty());
+
+
+	// calculate slack of remaining sink pins
+	for (int ipin : remaining_targets) {
+		/* Use criticality of 1. This makes all nets critical.  Note: There is a big difference between setting pin criticality to 0
+		compared to 1.  If pin criticality is set to 0, then the current path delay is completely ignored during routing.  By setting
+		pin criticality to 1, the current path delay to the pin will always be considered and optimized for */
+		if (!slacks) pin_criticality[ipin] = 1.0;

```

Comment:
I assume this only impacts the non-timing driven router.  It might be good to measure the performance with --timing_analysis off to see if this has any impact.